diff --git a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/blit_shaders/CMakeLists.txt b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/blit_shaders/CMakeLists.txt
index 33f016f722..d3c279befd 100644
--- a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/blit_shaders/CMakeLists.txt
+++ b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/blit_shaders/CMakeLists.txt
@@ -45,7 +45,7 @@
 cmake_minimum_required ( VERSION 3.7 )
 
 # Find Clang package and LLVM package
-find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
+# find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
 find_package(LLVM REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
 
 # Set the target devices
@@ -56,10 +56,10 @@ set (POSTFIX "9;1010;10;11;12")
 
 # If verbose output is enabled, print paths and target devices
 if(${CMAKE_VERBOSE_MAKEFILE})
-	get_property(clang_path TARGET clang PROPERTY LOCATION)
-	get_property(objcopy_path TARGET llvm-objcopy PROPERTY LOCATION)
-	message("Using clang from: ${clang_path}")
-	message("Using llvm-objcopy from: ${objcopy_path}")
+# 	get_property(clang_path TARGET clang PROPERTY LOCATION)
+# 	get_property(objcopy_path TARGET llvm-objcopy PROPERTY LOCATION)
+# 	message("Using clang from: ${clang_path}")
+# 	message("Using llvm-objcopy from: ${objcopy_path}")
 	message("Blit Shaders assembled for: ${TARGET_DEVS}")
 endif()
 
@@ -71,8 +71,8 @@ function(gen_kernel_bc TARGET_ID INPUT_FILE OUTPUT_FILE)
 	separate_arguments(CLANG_ARG_LIST UNIX_COMMAND "-x assembler -target amdgcn-amd-amdhsa -mcode-object-version=5 -fPIC -mcpu=${TARGET_ID} -o ${CODE_OBJECT} ${CMAKE_CURRENT_SOURCE_DIR}/${INPUT_FILE}")
 
 	# Add custom command to generate the kernel bitcode
-	add_custom_command(OUTPUT ${CODE_OBJECT} COMMAND clang ${CLANG_ARG_LIST}
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${INPUT_FILE} clang
+	add_custom_command(OUTPUT ${CODE_OBJECT} COMMAND /opt/rocm/bin/amdclang ${CLANG_ARG_LIST}
+	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${INPUT_FILE}
 	COMMENT "BUILDING bitcode for ${OUTPUT_FILE}..."
 	VERBATIM)
 
@@ -81,7 +81,7 @@ function(gen_kernel_bc TARGET_ID INPUT_FILE OUTPUT_FILE)
 	# Add custom command to extract binary from the bitcode
 	add_custom_command(OUTPUT ${OUTPUT_FILE}
 	COMMAND llvm-objcopy ${OBJCOPY_ARG_LIST}
-	DEPENDS ${CODE_OBJECT} llvm-objcopy
+	DEPENDS ${CODE_OBJECT}
 	COMMENT "Extracting binary for ${OUTPUT_FILE}..."
 	VERBATIM)
 
diff --git a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/CMakeLists.txt b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/CMakeLists.txt
index ba2be8ae58..e1b1cb516e 100644
--- a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/CMakeLists.txt
+++ b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/CMakeLists.txt
@@ -43,18 +43,18 @@
 cmake_minimum_required ( VERSION 3.7 )
 
 # Import target 'clang' and 'llvm-objcopy'
-find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
+# find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
 find_package(LLVM REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
 
-set (TARGET_DEVS "gfx900;gfx942;gfx950;gfx1010;gfx1030;gfx1100;gfx1200")
-set (POSTFIX "9;942;950;1010;10;11;12")
-set (SOURCE_SUFFIX ";;;;;;_gfx12")
+set (TARGET_DEVS "gfx900;gfx942;gfx1010;gfx1030;gfx1100;gfx1200")
+set (POSTFIX "9;942;1010;10;11;12")
+set (SOURCE_SUFFIX ";;;;;_gfx12")
 
 if(${CMAKE_VERBOSE_MAKEFILE})
-  get_property(clang_path TARGET clang PROPERTY LOCATION)
-  get_property(objcopy_path TARGET llvm-objcopy PROPERTY LOCATION)
-  message("Using clang from: ${clang_path}")
-  message("Using llvm-objcopy from: ${objcopy_path}")
+#   get_property(clang_path TARGET clang PROPERTY LOCATION)
+#   get_property(objcopy_path TARGET llvm-objcopy PROPERTY LOCATION)
+#   message("Using clang from: ${clang_path}")
+#   message("Using llvm-objcopy from: ${objcopy_path}")
   message("Trap handlers assembled for: ${TARGET_DEVS}")
 endif()
 
@@ -69,8 +69,8 @@ function(gen_kernel_bc TARGET_ID INPUT_FILE OUTPUT_FILE)
   "-x assembler -target amdgcn-amd-amdhsa -mcpu=${TARGET_ID} -o ${CODE_OBJECT} ${INPUT_FILE}")
 
   ## Add custom command to produce a code object file.
-  add_custom_command(OUTPUT ${CODE_OBJECT} COMMAND clang ${CLANG_ARG_LIST}
-    DEPENDS ${INPUT_FILE} clang
+  add_custom_command(OUTPUT ${CODE_OBJECT} COMMAND /opt/rocm/bin/amdclang ${CLANG_ARG_LIST}
+    DEPENDS ${INPUT_FILE}
     COMMENT "BUILDING bitcode for ${OUTPUT_FILE}..."
     VERBATIM)
 
@@ -79,7 +79,7 @@ function(gen_kernel_bc TARGET_ID INPUT_FILE OUTPUT_FILE)
   ## Extract .text segment
   add_custom_command(OUTPUT ${OUTPUT_FILE}
                      COMMAND llvm-objcopy ${OBJCOPY_ARG_LIST}
-                     DEPENDS ${CODE_OBJECT} llvm-objcopy
+                     DEPENDS ${CODE_OBJECT}
                      COMMENT "Extracting binary for ${OUTPUT_FILE}..."
                      VERBATIM)
 
diff --git a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/trap_handler_gfx12.s b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/trap_handler_gfx12.s
index 54236f368d..301870d0eb 100644
--- a/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/trap_handler_gfx12.s
+++ b/projects/rocr-runtime/runtime/hsa-runtime/core/runtime/trap_handler/trap_handler_gfx12.s
@@ -139,6 +139,7 @@
 .set HW_REG_SQ_PERF_SNAPSHOT_DATA1             , 0xf80f
 .set HW_REG_SQ_PERF_SNAPSHOT_DATA2             , 0xf810
 .set HW_REG_SQ_PERF_SNAPSHOT_DATA              , 0xf81b
+.set HW_REG_WAVE_SCHED_MODE                    , 0x1a
 
   // Macro to store the Correlation ID (Dispatch ID and Doorbell ID) into the current sample slot
   //
diff --git a/projects/rocr-runtime/runtime/hsa-runtime/image/blit_src/CMakeLists.txt b/projects/rocr-runtime/runtime/hsa-runtime/image/blit_src/CMakeLists.txt
index 93cb816de7..8714e8df33 100644
--- a/projects/rocr-runtime/runtime/hsa-runtime/image/blit_src/CMakeLists.txt
+++ b/projects/rocr-runtime/runtime/hsa-runtime/image/blit_src/CMakeLists.txt
@@ -43,7 +43,7 @@
 cmake_minimum_required ( VERSION 3.7 )
 
 # Import target 'clang'
-find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
+# find_package(Clang REQUIRED HINTS ${CMAKE_PREFIX_PATH}/llvm PATHS /opt/rocm/llvm )
 
 # Determine the target devices if not specified
 if (NOT DEFINED TARGET_DEVICES)
@@ -55,12 +55,12 @@ endif()
 set( TARGET_DEVICES ${TARGET_DEVICES} CACHE STRING "Build targets" FORCE )
 
 if(${CMAKE_VERBOSE_MAKEFILE})
-  get_property(clang_path TARGET clang PROPERTY LOCATION)
-  message("Using clang from: ${clang_path}")
+#   get_property(clang_path TARGET clang PROPERTY LOCATION)
+#   message("Using clang from: ${clang_path}")
   message("Build Setting:")
   message("  Target Devices*: ${TARGET_DEVICES}")
   message("  (Specify \";\" separated list of target IDs.)")
-  message("       Clang path: ${clang_path}")
+#   message("       Clang path: ${clang_path}")
 endif()
 
 ##==========================================
@@ -77,8 +77,8 @@ function(gen_kernel_bc TARGET_ID INPUT_FILE OUTPUT_FILE)
 
   ## Add custom command to produce a code object file.
   ## This depends on the kernel source file & compiler.
-  add_custom_command(OUTPUT ${OUTPUT_FILE} COMMAND clang ${CLANG_ARG_LIST}
-    DEPENDS ${INPUT_FILE} clang
+  add_custom_command(OUTPUT ${OUTPUT_FILE} COMMAND /opt/rocm/bin/amdclang ${CLANG_ARG_LIST}
+    DEPENDS ${INPUT_FILE}
     COMMENT "BUILDING bitcode for ${OUTPUT_FILE}..."
     VERBATIM)
 
